retail/interactive-tutorial/RetailSearch.Samples/SearchWithFacetSpec.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright 2021 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START retail_search_product_with_facet_spec]
+
+using Google.Cloud.Retail.V2;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+/// <summary>
+/// Search with facet spec sample class.
+/// </summary>
+public static class SearchWithFacetSpec
+{
+    /// <summary>The project number.</summary>
+    private static readonly string ProjectNumber = Environment.GetEnvironmentVariable("PROJECT_NUMBER");
```

Comment:
Does this need to be the project number or can it be the project ID. Most other samples use the project ID.

Reply:
Also, is there nothing we can do to have this as a parameter of the method shown in the sample? For instance, all on the same file:

```
[START region_tag]
public class SearchSample
{
    public SearchResult Search(string projectId) { ... }
}
[END region_tag]

public static class SearchTutorial
{
    [RetailSearch.Samples.Attributes.Example]
    public static SearchResult Search()
    {
          var projectId = Environment.GetEnvironmentVariable("PROJECT_NUMBER");
          var sample = new SearchSample();
          return sample.Search(projectId);
    }
}
```

Reply:
so that we can even keep the region tags and reuse the samples as the static documentation code samples, not only as an interactive tutorials? I think it is a good idea to combine them in such manner

Reply:
Suggested structure is applied